Starting point for event tracking and the `.on` interface (#38)
Financial-Times/o-comments: README.md

});
```

#### Interface

##### .on(eventName, callback)

- `eventName` - *required*
- `callBack` - *required*

The `.on` interface allows you to listen for [events](#events).

```js
const oComments = require('o-comments');
const Comments = new oComments();

Comments.on('component.render.successful', () => {
	console.log('The comments have rendered')
});

```

#### Events

> Events are omitted during key events and can be listened to using the [`.on` interface](#on).

Reviewer: ```suggestion
Events are emitted during key events and can be listened to using the [`.on` interface](#on).
```
omitted means "left out"/"ignored"

Author: 👍 Thanks Jake... I always get those mixed up

The naming of events uses a dot notation and follows a `category.action.state` namespacing. The categories and their events are listed below.

##### Component

> These events are anything to do with the component its self.

Reviewer: ```suggestion
These events are anything to do with the component itself.
```

> - **component.render.successful** - Omitted when the component has finished rendering and is ready for the user to interact with comments 

Reviewer: ```suggestion
- **component.render.successful** - Emitted when the component has finished rendering and is ready for the user to interact with comments 
```

##### Comment

These events are anything to do with comment interactions.

> - **comment.posted.successful** - Omitted when a users has successfully left a comment

Reviewer: ```suggestion
- **comment.posted.successful** - Emitted when a users has successfully left a comment
```

> - **comment.liked.successful** - Omitted when a users has liked a comment

Reviewer: ```suggestion
- **comment.liked.successful** - Emitted when a users has liked a comment
```

##### Auth

These events are anything to do with users being authenticated.

> - **auth.login.successful** - Omitted when a user has successfully logged in to Coral Talk

Reviewer: ```suggestion
- **auth.login.successful** - Emitted when a user has successfully logged in to Coral Talk
```

> - **auth.login.required** - Omitted when a logged out user has performed an action that requires them to login

Reviewer: ```suggestion
- **auth.login.required** - Emitted when a logged out user has performed an action that requires them to login
```
